src/github.com/getlantern/flashlight/ui/ui.go

```diff
@@ -112,13 +112,31 @@ func Show() {
 		if err != nil {
 			log.Errorf("Error opening page to `%v`: %v", uiaddr, err)
 		}
-		if externalUrl != "NO"+"_URL" && !openedExternal {
-			time.Sleep(4 * time.Second)
-			err = open.Run(externalUrl)
-			if err != nil {
-				log.Errorf("Error opening external page to `%v`: %v", uiaddr, err)
-			}
-			openedExternal = true
-		}
+		openExternalUrl()
+	}()
+}
+
+// openExternalUrl opens an external URL of one of our partners automatically
+// at startup if configured to do so. It should only open the first time in
+// a given session that Lantern is opened.
+func openExternalUrl() {
+	if openedExternal {
+		log.Debugf("Not opening external URL again")
+		return
+	}
+	defer func() {
```

Comment:
Hey I agree this should be cleaned up -- I can do that on some other pass @xiam.